internal/service/vmservice/bootstrap.go

```diff
@@ -69,13 +69,18 @@ func reconcileBootstrapData(ctx context.Context, machineScope *scope.MachineScop
 		return false, err
 	}
 
+	kubernetesVersion := ""
+	if machineScope.Machine.Spec.Version != nil {
+		kubernetesVersion = *machineScope.Machine.Spec.Version
+	}
+
 	machineScope.Logger.V(4).Info("reconciling BootstrapData.", "format", format)
 
 	// Inject userdata based on the format
 	if ptr.Deref(format, "") == ignition.FormatIgnition {
-		err = injectIgnition(ctx, machineScope, bootstrapData, biosUUID, nicData)
+		err = injectIgnition(ctx, machineScope, bootstrapData, biosUUID, nicData, kubernetesVersion)
```

Comment:
This line has no tests.

Reply:
I have added tests that cover this line. It took some time and I had to make some changes.
The whole ignition bootstrapping had no real tests, I think this should have been done before.
Anyway, now we are at 88.9% coverage of the new code. As I don't have access to the SonarQube reports, I'm not sure which lines are still not covered.

Reply:
Indeed it should have been done before :)